engines/build.info

```diff
@@ -1,4 +1,4 @@
-IF[{- !$disabled{"engine"} -}]
+IF[{- !$disabled{"engine"} && !$disabled{'deprecated-3.0'} -}]
   $PADLOCKASM=
```

Comment:
How is the padlock engine related to RSA to such a degree that you disable it entirely?

Reply:
It is disabling all engines, not just the padlock engine.  With the various EVP_*_meth_ functions deprecated, engines aren't much use.  I'm only hitting this here because I didn't deprecate the meth calls for digests and ciphers.

Reply:
The `_meth_` functions used in e_padlock.c haven't been deprecated yet as far as I know, as padlock is cipher only.  The same goes for the other engines.  This is much too broad a stroke.

Reply:
There is an intention to deprecate these for 3.0 isn't there?  That was my understanding.

Reply:
I would prefer if we left that to the future PR that does that work

Reply:
Sigh...

I'll put together a PR to deprecate all of the `_meth_` functions.

Reply:
Yes, "disabling low-level RSA disables all engines" is not good.

Reply:
I'm only disabling the _dasync_ engine now.

Reply:
I still think this is not acceptable.   If you can't disable the low-level RSA functions without *leaving engines intact* than don't disable in this release.

(Of course, I question the overall worth of this and how important it is given the FIPS goal but you knew I was going to raise that. :)

Reply:
A decision was recently made to disable command line applets instead of converting them to avoid the low-level calls.  Why should engines be more worthy of such conversion?

The low-level functions are only deprecated, engines can still be built using them.  If the build includes the deprecated functions, the engine will be built.  It is disabled only in a `no-deprecated` build.

Reply:
Engines are essentially intact...  except for the future prediction called `no-deprecated`.

Reply:
Apart from the earlier changes that needed to be made for the cipher & digest low level deprecation....

Reply:
One of the deprecated apps didn't require any changes to work -- it wasn't using any deprecated called.  It still went.

Reply:
It's also a demonstration engine that's been disabled.

Reply:
The difference is that with no-deprecated there are alternatives.  There are no available alternatives to engines.

This is really wrong-headed.  You are spending times on things that some in the project want to do, but that aren't necessary to a release that has already slipped once and is even now still seen as "aggressive."  And yet, you won't spend time on things people want, like the QUIC PR, because it's "not strictly required for FIPS."  Well, heck, neither is this; it's admittedly a nice to have.

So instead, the project is doing things that *almost nobody will turn on* and not doing things *everyone will enable.*

I really hope this doesn't turn OpenSSL into becoming a niche TLS offering.  Seriously.

Reply:
No, the non-deprecated thing for engines are (will be) providers. I have to repeat - if we intend to remove the engines support at some point it is better to deprecate it ASAP to signalize the intent clearly.
The no-deprecated build should not be normally used. Clearly the no-deprecated is not to be used in general purpose builds shared among various applications such as the system openssl library in an OS distribution.